docs/source/optimization.rst

```diff
@@ -0,0 +1,140 @@
+.. module:: statsmodels.base.optimizer
+.. currentmodule:: statsmodels.base.optimizer
+
+Optimization
+============
+
```

Comment:
I think a preliminary sentence about non-scipy optimization would be helpful
e.g. something like

"statsmodels uses three types of algorithms for the estimation of the parameters of a model. Linear models such as OLS are directly estimated using appropriate linear algebra functions. Some models use specific algorithm such as iteratively reweighted least squares in GLM and RLM, or the most common method for nonlinear optimization is by using the optimizers in scipy/"

Reply:
Sure. You'll have to help me if you want to go through all of them. I don't know much (anything) on time series and all the modules there and I think that's pretty scipy-heavy, too.

Reply:
I just want us to mention the type of optimization methods without providing a full list of models.

(actually it's not that difficult, skipping some models in the sandbox
linear regression OLS/WLS/GLS and VAR and VECM have closed for linalg estimators, 
only RLM and GLM have IRLS, 
all others have only scipy optimizers + our newton.
GLM is currently the only one with irls and scipy optimizers.

There are some hybrid versions like GLSAR that we don't mention.

... maybe there would be more in a full list, e.g. fit_regularized in OLS/WLS and in GLM uses coordinate descend. 
)